### ISSUE TITLE
WebSockets Next: provide strategies to process unhandled failures

### ISSUE THREAD

**Issue by mkouba** [quarkusio/quarkus#40648]
### Description

Right now, we log an error message when a unhandled failure occurs (unless it's a  "WebSocket closed" error in which case we only log a debug message).

It seems reasonable to provide several strategies instead:

1. Close the connection (default)
2. Log an error message
3. NOOP

**Comment by sberyozkin** [comment]
It would nice to have an option to redirect a user to a configured page once the connection is closed

**Comment by mkouba** [comment]
> It would nice to have an option to redirect a user to a configured page once the connection is closed

I don't think that's possible. There's no HTTP response once the upgrade is finished. We could use a specific [close reason](https://datatracker.ietf.org/doc/html/rfc6455#autoid-43) though.

**Comment by mkouba** [comment]
> > It would nice to have an option to redirect a user to a configured page once the connection is closed
> 
> I don't think that's possible. There's no HTTP response once the upgrade is finished. We could use a specific [close reason](https://datatracker.ietf.org/doc/html/rfc6455#autoid-43) though.

I mean not possible after the upgrade but it should be possible if we detect a problem during the handshake, e.g. once we implement #40622.

However, this issue is more generic, not specifically related to security.

**Comment by sberyozkin** [comment]
>I mean not possible after the upgrade but it should be possible if we detect a problem during the handshake, e.g. once we implement https://github.com/quarkusio/quarkus/issues/40622.

Sure

**Comment by mkouba** [comment]
> > I mean not possible after the upgrade but it should be possible if we detect a problem during the handshake, e.g. once we implement #40622.
> 
> maybe I should had commented on #40622, apologies.

No problem. Speaking of "eagerly performed `@OnOpen`" - I'm afraid that this would complicate the workflow a lot. So I think that we should rather consider the case where security annotations are declared on the endpoint class and applied before the endpoint instance is actually created.
